spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/web/reactive/ReactiveMultipartAutoConfigurationTests.java

```diff
@@ -0,0 +1,251 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.runner.ReactiveWebApplicationContextRunner;
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner;
+import org.springframework.boot.web.codec.CodecCustomizer;
+import org.springframework.core.annotation.AnnotationAwareOrderComparator;
+import org.springframework.http.codec.CodecConfigurer;
+import org.springframework.http.codec.multipart.DefaultPartHttpMessageReader;
+import org.springframework.http.codec.support.DefaultServerCodecConfigurer;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.unit.DataSize;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link ReactiveMultipartAutoConfiguration}.
+ *
+ * @author Chris Bono
+ */
+class ReactiveMultipartAutoConfigurationTests {
+
+	private final ReactiveWebApplicationContextRunner contextRunner = new ReactiveWebApplicationContextRunner()
+			.withConfiguration(AutoConfigurations.of(ReactiveMultipartAutoConfiguration.class));
+
+	private static final boolean DEFAULT_STREAMING = false;
+
+	private static final int DEFAULT_MAX_IN_MEMORY_SIZE = 256 * 1024;
+
+	private static final int DEFAULT_MAX_PARTS = -1;
+
+	private static final int DEFAULT_MAX_HEADERS_SIZE = 8 * 1024;
+
+	private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+
+	private static final long DEFAULT_MAX_DISK_USAGE_PER_PART = -1;
+
+	private static final Path DEFAULT_FILE_STORAGE_DIRECTORY = Paths.get(System.getProperty("java.io.tmpdir"),
+			"spring-multipart");
+
+	@Test
+	void shouldNotProvideCustomizerForNonReactiveApp() {
+		new WebApplicationContextRunner()
+				.withConfiguration(AutoConfigurations.of(ReactiveMultipartAutoConfiguration.class))
+				.run((context) -> assertThat(context).doesNotHaveBean(CodecCustomizer.class));
+	}
+
+	@Test
+	void shouldNotProvideCustomizerWhenDefaultPartHttpMessageReaderNotAvailable() {
+		this.contextRunner.withClassLoader(new FilteredClassLoader(DefaultPartHttpMessageReader.class))
+				.run((context) -> assertThat(context).doesNotHaveBean(CodecCustomizer.class));
+	}
+
+	@Test
+	void shouldNotProvideCustomizerWhenCodecConfigurerNotAvailable() {
+		this.contextRunner.withClassLoader(new FilteredClassLoader(CodecConfigurer.class))
+				.run((context) -> assertThat(context).doesNotHaveBean(CodecCustomizer.class));
+	}
+
+	@Test
+	void customizerSetsAppropriatePropsWhenStreamingEnabled() {
+		this.contextRunner.withPropertyValues("spring.webflux.multipart.streaming:true",
+				"spring.webflux.multipart.max-in-memory-size=1GB", "spring.webflux.multipart.max-headers-size=512MB",
```

Comment:
My DRY alarm is going off... however, I think the repetition is just at the cusp of acceptable. I made a pass at factoring out some of the common things between the tests but their slight differences are just enough that it was not an improvement (it was less readable).